src/lava/lib/optimization/solvers/generic/builder.py

```diff
@@ -140,55 +142,73 @@ def _create_process_constructor(
 
         def constructor(
             self,
+            backend,
             hyperparameters: ty.Dict[str, ty.Union[int, npt.ArrayLike]],
             name: ty.Optional[str] = None,
             log_config: ty.Optional[LogConfig] = None,
         ) -> None:
             super(type(self), self).__init__(
+                backend=backend,
                 hyperparameters=hyperparameters,
                 name=name,
                 log_config=log_config,
             )
             self.problem = problem
             self.hyperparameters = hyperparameters
+            self.backend = backend
+            self.is_continuous = 0
+            self.is_discrete = 0
             if not hasattr(problem, "variables"):
                 raise Exception(
                     "An optimization problem must contain " "variables."
                 )
-            if hasattr(problem.variables, "continuous") or isinstance(
-                problem.variables, ContinuousVariables
+            if (
+                hasattr(problem.variables, "continuous")
+                and problem.variables.continuous.num_variables is not None
             ):
                 self.continuous_variables = Var(
-                    shape=(problem.variables.continuous.num_vars, 2)
+                    shape=(problem.variables.continuous.num_variables,)
                 )
-            if hasattr(problem.variables, "discrete") or isinstance(
-                problem.variables, DiscreteVariables
+                self.is_continuous = 1
+            if (
+                hasattr(problem.variables, "discrete")
+                and problem.variables.discrete.num_variables is not None
             ):
                 self.discrete_variables = Var(
                     shape=(
-                        problem.variables.num_variables,
+                        problem.variables.discrete.num_variables,
                         # problem.variables.domain_sizes[0]
                     )
                 )
+                self.is_discrete = 1

```

Comment:
Hm... If some variables are discrete and some are not, self.is_discrete would be 1. But would that correctly represent whether it's a concrete solver?

Reply:
we tackle mixed problems when we have a mixed problems solver